dbt/include/trino/macros/materializations/materialized_view.sql

```diff
@@ -0,0 +1,41 @@
+{% materialization materialized_view, adapter="trino" %}
+  {%- set target_relation = this %}
+  {%- set existing_relation = load_relation(this) -%}
+
+  {{ run_hooks(pre_hooks) }}
+
+  {% if existing_relation is none %}
+    {{ log("No existing materialized view found, creating materialized view...", info=true) }}
+    {%- set build_sql = build_materialized_view(target_relation) %}
+
+  {% elif existing_relation.type != "materializedview" %}
+    {{ log("Found a " ~ existing_relation.type ~ " with same name. Dropping it...", info=true) }}
+    {{ adapter.drop_relation(existing_relation) }}
+    {%- set build_sql = build_materialized_view(target_relation) %}
+
+  {% elif existing_relation.type == "materializedview" %}
+    {{ log("Refreshing materialized view '" ~ existing_relation.identifier ~ "'...", info=true) }}
+    {%- set build_sql = refresh_materialized_view(target_relation) %}
```

Comment:
Why refresh as pre action. How would you know if the SQL of the MV is the same.

Reply:
I refactored it to be similar as in [dbt experimental mv](https://github.com/dbt-labs/dbt-labs-experimental-features/blob/main/materialized-views/macros/default/materialized_view.sql), so mv is either created or refreshed depending on the [full_refresh](https://docs.getdbt.com/reference/resource-configs/full_refresh) config/flag